xcube/core/gen2/opener.py

```diff
@@ -0,0 +1,98 @@
+                 store_pool: DataStorePool = None):
+        self._input_configs = input_configs
+        self._cube_config = cube_config
+        self._store_pool = store_pool
+
+    def open_cubes(self) -> Sequence[xr.Dataset]:
+        cubes = []
+        with observe_progress('Opening input(s)', len(self._input_configs)) as progress:
+            for input_config in self._input_configs:
+                cubes.append(self._open_cube(input_config))
+                progress.worked(1)
+        return cubes
+
+    def _open_cube(self, input_config: InputConfig) -> xr.Dataset:
+        all_cube_params = self._cube_config.to_dict()
+        opener_id = input_config.opener_id
+        store_params = input_config.store_params or {}
+        open_params = input_config.open_params or {}
+        normalisation_required = False
+        if input_config.store_id:
+            store_instance = get_data_store_instance(input_config.store_id,
+                                                     store_params=store_params,
+                                                     store_pool=self._store_pool)
+            store = store_instance.store
+            if opener_id is None:
+                opener_ids = None
+                type_specifiers = store.get_type_specifiers_for_data(input_config.data_id)
+                for type_specifier in type_specifiers:
+                    if TYPE_SPECIFIER_CUBE.is_satisfied_by(type_specifier):
+                        opener_ids = \
+                            store.get_data_opener_ids(data_id=input_config.data_id,
+                                                      type_specifier=type_specifier)
+                        break
+                if not opener_ids:
+                    for type_specifier in type_specifiers:
+                        if TYPE_SPECIFIER_DATASET.is_satisfied_by(type_specifier):
+                            opener_ids = \
+                                store.get_data_opener_ids(data_id=input_config.data_id,
+                                                          type_specifier=type_specifier)
+                            normalisation_required = True
+                            break
+                if not opener_ids:
+                    raise DataStoreError(f'Data store "{input_config.store_id}" '
+                                         f'does not support datasets')
+                opener_id = opener_ids[0]
+            opener = store
+            open_params.update(opener_id=opener_id, **open_params)
+        else:
+            opener = new_data_opener(opener_id)
+            open_params.update(**store_params, **open_params)
+        open_params_schema = opener.get_open_data_params_schema(input_config.data_id)
+        cube_params = {k: v for k, v in all_cube_params.items() if k in open_params_schema.properties}
+        cube = opener.open_data(input_config.data_id, **open_params, **cube_params)
+        if normalisation_required:
+            cube = normalize_dataset(cube)
```

Comment:
Actually, we cannot be certain that the dataset is a cube after the normalisation. I think it would be better to use verify_cube here and return nothing in case the verification fails.
I realize this issue existed before your PR and therefore is not really part of it, but I just happen to notice it whilst I was reviewing your changes.

Reply:
This is your code.

Reply:
I acknowledge that. What do you suggest we should do with the issue, though?
1. Ignore?
2. Handle here?
3. Handle elsewhere?

Reply:
> Actually, we cannot be certain that the dataset is a cube after the normalisation.

Why not? Normalisation may just fail or do nothing at all.

Reply:
Therefore:

1. Ignore for time being until we face a new blocker.